### DISCUSSION TITLE
Why FlatePickr return undefine when submit form

### DISCUSSION THREAD

**Discussion by abdulmateentechbits** [react-hook-form/react-hook-form#9262] [Q&A] [answered]
When i submit the form flatpickr return undefined
https://codesandbox.io/s/demo-forked-qltj2v?file=/src/components/forms/datePicke.jsx

**Comment by abdulmateentechbits** [comment]
@leapful 
@blabute

**Comment by abdulmateentechbits** [comment]
@bluebill1049

**Comment by Moshyfawn** [comment, marked as answer]
The `Flatpickr` component you're integrating with RHF exposes non-web statndart props / callbacks, so you can't use the `register` method with it.

You need to use the `Flatpickr` component as a controlled input with `Controller` / `useController` API. Here's an example:
```js
const { control } = useForm({
  defaultValues: {
    sells_date: new Date().toLocaleDateString("en-GB")
  }
})

return (
  <Controller
    control={control}
    name="sells_date"
    render={({ field: { onChange, ...fieldProps } }) => (
      <Flatpickr
        {...fieldProps}
        onChange={(dates, currentdateString) => onChange(currentdateString)}
        data-enable-time
        className="form-control"
        options={{ dateFormat: "d/m/Y" }}
      />
    )}
  />
)
```

Here's [a working codesandbox](https://codesandbox.io/s/demo-forked-hc992j?file=%2Fsrc%2Fcomponents%2Fforms%2FdatePicke.jsx%3A738-1092).

_P.S. Please, avoid spam-tagging team members, @abdulmateentechbits_

**Reply by abdulmateentechbits** [reply]
Thanks for your feedback !
Its work for me.
Ok  I am new to the discussion group thanks for help me it was really painful to me
